File: python/MaxRevAd.py

def greedy(n, a, b):
  sum = 0
  for i in range(n):
    sum += a[i]*b[i]
  return sum


n = int(input())
a = [*map(int, input().split())]
b = [*map(int, input().split())]

a.sort(reverse=True)
b.sort(reverse=True)
print(greedy(n, a, b))

'''
file = open('sample/3_3_dot_product20180216.in', 'r')
n = int(file.readline())
a = [*map(int, file.readline().split())]
b = [*map(int, file.readline().split())]
a.sort(reverse=True)
b.sort(reverse=True)
print(greedy(n, a, b))
'''

File: c++/MaxRevAd.cpp
#include "stdafx.h"

using namespace std;

bool comp(long long a, long long b)
{
	return a > b;
}

long long greedy(int n, vector<long long> &a, vector<long long> &b)
{
	long long sum = 0;
	
	for (int i = 0; i < n; i++)
		sum += a[i] * b[i];

	return sum;
}

int start()
{
	/*int n;
	cin >> n;

	vector<long> a;
	vector<long> b;
	long tmp;

	for (int i = 0; i < n; i++) {
		cin >> tmp;
		a.push_back(tmp);
	}

	for (int i = 0; i < n; i++) {
		cin >> tmp;
		b.push_back(tmp);
	}

	sort(a.begin(), a.end(), comp);
	sort(b.begin(), b.end(), comp);

	cout << greedy(n, a, b) << endl;*/

	ifstream file;
	file.open("C:/Users/sohee-um/program/workspace/etc/resource/py/sample/3_3_dot_product20180216.in");
	if (!file) {
		cout << "Unable to open file!" << endl;
		exit(1);
	}

	int n;
	file >> n;

	vector<long long> a;
	vector<long long> b;
	long long tmp;

	for (int i = 0; i < n; i++) {
		file >> tmp;
		a.push_back(tmp);
	}

	for (int i = 0; i < n; i++) {
		file >> tmp;
		b.push_back(tmp);
	}

	sort(a.begin(), a.end(), comp);
	sort(b.begin(), b.end(), comp);

	cout << greedy(n, a, b) << endl;

	return 0;
}
